spec/models/dialog_spec.rb

```diff
@@ -386,6 +386,41 @@
       end
     end
 
+    context "non-circular field associations" do
+      before do
+        dialog.dialog_tabs << FactoryGirl.create(:dialog_tab, :label => 'tab')
+        dialog.dialog_tabs.first.dialog_groups << FactoryGirl.create(:dialog_group, :label => 'group')
+        dialog.dialog_tabs.first.dialog_groups.first.dialog_fields << FactoryGirl.create(:dialog_field, :label => 'field 1', :name => 'field1')
+        dialog.dialog_tabs.first.dialog_groups.first.dialog_fields << FactoryGirl.create(:dialog_field, :label => 'field 2', :name => 'field2')
+        FactoryGirl.create(:dialog_field_association, :trigger_id => DialogField.find_by(:name => 'field1').id, :respond_id => DialogField.find_by(:name => 'field2').id)
+      end
+
+      it "adds error on simple circular reference" do
+        FactoryGirl.create(:dialog_field_association, :trigger_id => DialogField.find_by(:name => 'field2').id, :respond_id => DialogField.find_by(:name => 'field1').id)
+
+        expect { dialog.save! }
+          .to raise_error(ActiveRecord::RecordInvalid, /Dialog field associations cannot be circular/)
```

Comment:
No need to multi-line this.  We don't enforce the string length limit and the lines above are much longer.